podman_compose.py

```diff
@@ -1308,14 +1339,16 @@ def flat_deps(services, with_extends=False):
                 if ext != name:
                     deps.add(ext)
                 continue
-        deps_ls = srv.get("depends_on", None) or []
-        if isinstance(deps_ls, str):
```

Comment:
I think we should support `str` still and raise a warning, otherwise we will break a number of configurations already using this code path.